beacon_node/lighthouse_network/src/peer_manager/peerdb/sync_status.rs

```diff
@@ -28,6 +28,19 @@ pub struct SyncInfo {
     pub earliest_available_slot: Option<Slot>,
 }
 
+impl SyncInfo {
+    /// Returns true if the provided slot is greater than `earliest_available_slot`.
+    ///
+    /// If `earliest_available_slot` does is None, then we just assume that the peer has the slot.
```

Comment:
```suggestion
    /// If `earliest_available_slot` is None, then we just assume that the peer has the slot.
```

Reply:
This can only be `None` if the peer does not support StatusV2 right?
I guess we haven't formally deprecated V1 yet, so this behaviour is probably ok?

Reply:
This is essentially the behaviour we need to exhibit pre v2. 
Basically, we are saying "tell me what your earliest available slot is or i'll assume its backfilled all the way its supposed to be"